ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java

```diff
@@ -178,6 +182,12 @@ String toString(RaftGroupId groupId, CompletableFuture<RaftServerImpl> f) {
     this.factory = ServerFactory.cast(rpcType.newFactory(parameters));
 
     this.serverRpc = factory.newRaftServerRpc(this);
+
+    // TODO: Support multi-raft and should pass StateMachineRegistry to DataStreamServerImpl instead of StateMachine
```

Comment:
Since RaftServer already supports multi-raft, DataStreamServer should simply invoke the RaftServer API. Then, it will automatically support multi-raft.  In some future changes, we should not even pass StateMachine in the DataStreamServerImpl constructor.  The StateMachine should be obtained from RaftServer.

Reply:
@szetszwo Got it, Thanks for explanation.